utils/slice.go

```diff
@@ -19,6 +22,24 @@ func IndexOf(strs []string, str string) int {
 	return -1
 }
 
+// ValidateString checks if the given string is entirely composed of utf8 runs and if its normalized as NFKC
+func ValidateString(str string, canBeEmpty bool) error {
```

Comment:
For extra safety we should only expose two functions one that doesn't allow empty strings and one that does (if needed at all, I don't know where it would be) to avoid an incorrect bool being passed unnoticed.

Reply:
good point